File: schemas/metrics.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel
from .asset_common import Plataforma


class MetricBase(BaseModel):
    plataforma: Plataforma
    views: Optional[int] = None
    likes: Optional[int] = None
    comments: Optional[int] = None
    shares: Optional[int] = None
    engagement: Optional[int] = None
    reach: Optional[int] = None
    raw_json: Optional[dict] = None


class MetricCreate(MetricBase):
    pass


class MetricOut(MetricBase):
    id: int
    asset_id: int
    capturado_em: datetime

    class Config:
        from_attributes = True

File: frontend/src/lib/metrics.ts
import { api } from "./api";

export type Metric = {
  id: number;
  asset_id: number;
  plataforma:
    | "site"
    | "youtube"
    | "instagram"
    | "tiktok"
    | "kwai"
    | "youtube_shorts"
    | "facebook";
  capturado_em: string;
  views?: number | null;
  likes?: number | null;
  comments?: number | null;
  shares?: number | null;
  engagement?: number | null;
  reach?: number | null;
  raw_json?: Record<string, unknown> | null;
};

export async function listMetrics(limit = 25): Promise<Metric[]> {
  const { data } = await api.get<Metric[]>("/metrics");
  const ordered = [...data].sort(
    (a, b) =>
      new Date(b.capturado_em).getTime() -
      new Date(a.capturado_em).getTime(),
  );
  return ordered.slice(0, limit);
}
